### PR TITLE
test(docker): cover adapter edge cases identified in #610

### LINKED ISSUE #610: Docker SDK adapter: coverage gaps in service adapters (nil/empty/error paths)
Surfaced during the audit of [#605](https://github.com/netresearch/ofelia/issues/605) / [#606](https://github.com/netresearch/ofelia/pull/606).

The audit identified a cluster of low-severity but real coverage gaps in `core/adapters/docker/` that would help us catch regressions in the service adapters before they ship. Bundling them as a single issue rather than separate ones because the test fixtures are similar.

## Gaps

1. **`Close()` after partial construction failure** — no test exercises `Close()` when `NegotiateAPIVersion` failed but the SDK was already constructed. No idempotency test for double-`Close()` either.
2. **`ContainerServiceAdapter.Inspect("")`** (`container.go:75`) — empty ID, untested. Likely returns an SDK error, but the contract should be pinned.
3. **`ExecServiceAdapter.Create(ctx, "", nil)`** (`exec.go:25`) — nil `ExecConfig` is dereferenced; could panic. Untested.
4. **`ExecServiceAdapter.Run` with nil `stdout`/`stderr` and non-TTY** (`exec.go:111`) — `stdcopy.StdCopy(nil, nil, ...)` likely errors or panics. Untested.
5. **`event.go::Subscribe` error channel is buffered to 1** — two consecutive errors from the SDK block the producer forever. Untested.
6. **`NewClient()` vs `NewClientWithConfig(DefaultConfig())` parity** — no test pins them as equivalent. A future refactor splitting the paths could regress.
7. **`TestClientConfigHTTPSHost` and `TestClientConfigTCPHost`** at `client_test.go:154-172` only assert struct-field assignment — they don't exercise `createHTTPClient` and contribute nothing beyond a passing line. Either delete or upgrade to assert transport shape.
8. **Invalid `DOCKER_HOST` strings** (`unix://` empty path, `tcp://` no port, scheme-only, `=`, `not-a-url`) — `resolveDockerHost` is a pure pass-through with no validation. The bad value only surfaces at first request, not at construction. Related to the scheme-validation issue.

## Suggested approach

- Add a small `internal/dockertest` helper (or export `NewClientFromSDK`) so adapters can be unit-tested against a stub SDK without needing a real Docker daemon. This unblocks #2–#5 cleanly.
- Table-driven tests per adapter for the empty-ID / nil-config / canceled-ctx / double-Close edges.

## Severity

Hardening — not regression-risk today, but each of these has the shape of a bug we'd catch in CI if the test existed. Several have nil-deref or hang potential.

### PR DESCRIPTION
**Pull request by CybotTM**
## Summary

Closes the eight test-coverage gaps in `core/adapters/docker/` identified in #610 (audit fallout from #605/#606).

Tests follow strict TDD discipline: every new test that required a production fix was authored first, observed failing, and then guarded.

## Gaps closed

- **#1 — Close() idempotency** (`TestClient_CloseIdempotent`): pure pinning. SDK already idempotent.
- **#2 — `Inspect("")`** (`TestContainerServiceAdapter_Inspect_EmptyID`): pure pinning. The SDK validates empty IDs locally with `client.emptyIDError`; no daemon required.
- **#3 — `Create(ctx, ID, nil)`** (`TestExecServiceAdapter_Create_NilConfig`): **production fix** — empirically panicked on `config.User`. Now returns `ErrNilExecConfig`.
- **#4 — `Run` with nil writers in non-TTY** (`TestExecServiceAdapter_Run_NilWritersNonTTY`): **production fix** — `stdcopy.StdCopy(nil, nil, ...)` panicked. Now returns `ErrNoExecOutputWriter`. TTY mode still tolerates nil stdout (consistent with the existing skip in that branch).
- **#5 — Subscribe error channel buffer-of-1** (`TestEventServiceAdapter_Subscribe_ExitsOnError` + `..._ExitsOnContextCancel`): **investigated, no production change**. The hypothesised "two consecutive errors" cannot block the producer because the goroutine `return`s after the first error. Pinned that exit contract via `httptest` stub. Companion test pins clean cancellation.
- **#6 — `NewClient()` vs `NewClientWithConfig(DefaultConfig())` parity** (`TestNewClient_DelegatesToConfigDefault`): pure pinning of equivalent transport shape.
- **#7 — Dead `TestClientConfigHTTPSHost` / `TestClientConfigTCPHost`** at `client_test.go:154-172`: deleted; replaced by `TestCreateHTTPClient_TransportShape`, a table-driven test that asserts `DialContext` presence and `ForceAttemptHTTP2` per host scheme using loopback addresses.
- **#8 — Invalid `DOCKER_HOST` strings**: out of scope per task brief (validation tracked separately, related to #609). No changes here.

## Production changes

Only [`core/adapters/docker/exec.go`](https://github.com/netresearch/ofelia/blob/fix/issue-610/core/adapters/docker/exec.go):

- New sentinel errors `ErrNilExecConfig`, `ErrNoExecOutputWriter`.
- `Create` and `Run` now reject the panicking inputs at the top of the method.

All other commits are pure tests.

## Commit layout

- [`98ec26b`](https://github.com/netresearch/ofelia/commit/98ec26b) `fix(docker): guard ExecServiceAdapter against nil config and nil writers`
- [`8b365f4`](https://github.com/netresearch/ofelia/commit/8b365f4) `test(docker): pin ContainerServiceAdapter.Inspect empty-ID contract`
- [`627f365`](https://github.com/netresearch/ofelia/commit/627f365) `test(docker): pin EventServiceAdapter.Subscribe exit behavior`
- [`e04eb95`](https://github.com/netresearch/ofelia/commit/e04eb95) `test(docker): cover Close idempotency, transport shape, client parity`
- [`c140ae7`](https://github.com/netresearch/ofelia/commit/c140ae7) `chore(docker): satisfy err113/intrange/misspell lint after #610 tests`

## Test plan

- [x] `go test -race -count=3 ./core/adapters/docker/` passes
- [x] `golangci-lint run --timeout=3m ./core/adapters/docker/...` clean (0 issues)
- [x] `go test -count=1 ./...` passes (full suite)
- [x] `go build ./...` clean
- [x] Each failing test (#3, #4) observed failing pre-fix
- [x] Each test uses loopback addresses; no real Docker daemon required
- [ ] CI green

Fixes #610

### CODE REVIEW

File: core/adapters/docker/event_test.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+		for range errCh {
+		}
+	}()
+
+	select {
+	case <-done:
+	case <-time.After(3 * time.Second):
+		t.Fatal("Subscribe goroutine did not exit within 3s of ctx cancel")
+	}
+}
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply core/adapters/docker/event_test.go:143]: Addressed in commit bd8e77a: switched nil-config/no-writer tests to errors.Is on the new sentinels, extracted newLoopbackSDKClient helper to drop the duplicate SDK-setup boilerplate flagged by SonarCloud, and added a CHANGELOG entry for the two new sentinel errors.

File: core/adapters/docker/exec.go [hunk 2]
```diff
@@ -21,8 +21,23 @@ type ExecServiceAdapter struct {
...
+// Sentinel errors for input validation. Defined as static values so callers
+// can use errors.Is for branching and to satisfy err113 lint guidance.
+var (
+	// ErrNilExecConfig is returned when an ExecConfig pointer is nil.
+	ErrNilExecConfig = errors.New("exec: nil ExecConfig")
+	// ErrNoExecOutputWriter is returned when Run is invoked in non-TTY mode
+	// with both stdout and stderr nil. stdcopy.StdCopy panics in that case
+	// the moment it has output to dispatch.
+	ErrNoExecOutputWriter = errors.New("exec: non-TTY mode requires at least one of stdout or stderr")
+)
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply core/adapters/docker/exec.go:33]: Addressed in commit bd8e77a: switched nil-config/no-writer tests to errors.Is on the new sentinels, extracted newLoopbackSDKClient helper to drop the duplicate SDK-setup boilerplate flagged by SonarCloud, and added a CHANGELOG entry for the two new sentinel errors.

File: core/adapters/docker/exec.go [hunk 3]
```diff
@@ -84,6 +99,17 @@ func (s *ExecServiceAdapter) Inspect(ctx context.Context, execID string) (*domai
...
+	if config == nil {
+		return -1, ErrNilExecConfig
+	}
+	// Non-TTY mode demultiplexes via stdcopy.StdCopy, which panics if both
+	// writers are nil and there is any output to dispatch. Guard the input
+	// to keep the contract symmetric with TTY mode (which already tolerates
+	// a nil stdout by skipping the copy).
+	if !config.Tty && stdout == nil && stderr == nil {
+		return -1, ErrNoExecOutputWriter
+	}
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply core/adapters/docker/exec.go:111]: Addressed in commit bd8e77a: switched nil-config/no-writer tests to errors.Is on the new sentinels, extracted newLoopbackSDKClient helper to drop the duplicate SDK-setup boilerplate flagged by SonarCloud, and added a CHANGELOG entry for the two new sentinel errors.

File: core/adapters/docker/exec_test.go [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	cfg := &domain.ExecConfig{Cmd: []string{"true"}, Tty: false}
+	// Non-TTY mode: stdcopy demuxing path is exercised.
+	code, err := adapter.Run(context.Background(), "some-container", cfg, nil, nil)
+	if err == nil {
+		t.Fatal("expected error when both writers are nil in non-TTY mode, got nil")
+	}
+	if code != -1 {
+		t.Errorf("expected exit code -1 on error, got %d", code)
+	}
+}
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply core/adapters/docker/exec_test.go:58]: Addressed in commit bd8e77a: switched nil-config/no-writer tests to errors.Is on the new sentinels, extracted newLoopbackSDKClient helper to drop the duplicate SDK-setup boilerplate flagged by SonarCloud, and added a CHANGELOG entry for the two new sentinel errors.

File: core/adapters/docker/exec_test.go [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+	if err == nil {
+		t.Fatal("expected error for nil config, got nil")
+	}
+	if id != "" {
+		t.Errorf("expected empty exec ID on error, got %q", id)
+	}
+	if !strings.Contains(strings.ToLower(err.Error()), "config") &&
+		!strings.Contains(strings.ToLower(err.Error()), "nil") {
+		t.Errorf("expected error mentioning nil/config, got: %v", err)
+	}
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply core/adapters/docker/exec_test.go]: Addressed in commit bd8e77a: switched nil-config/no-writer tests to errors.Is on the new sentinels, extracted newLoopbackSDKClient helper to drop the duplicate SDK-setup boilerplate flagged by SonarCloud, and added a CHANGELOG entry for the two new sentinel errors.

File: core/adapters/docker/event_test.go [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+	t.Cleanup(func() { _ = sdk.Close() })
+
+	adapter := &EventServiceAdapter{client: sdk}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	eventCh, errCh := adapter.Subscribe(ctx, domain.EventFilter{})
+
+	// Let the goroutine wire up, then cancel.
+	time.Sleep(50 * time.Millisecond)
+	cancel()
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply core/adapters/docker/event_test.go:127]: Addressed in commit bd8e77a: switched nil-config/no-writer tests to errors.Is on the new sentinels, extracted newLoopbackSDKClient helper to drop the duplicate SDK-setup boilerplate flagged by SonarCloud, and added a CHANGELOG entry for the two new sentinel errors.

File: core/adapters/docker/client_internal_test.go [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+			}
+		})
+	}
+}
+
+// TestNewClient_DelegatesToConfigDefault pins that NewClient() and
+// NewClientWithConfig(DefaultConfig()) produce equivalent transport
+// shape, so a future refactor splitting the two paths cannot regress
+// the contract silently.
+//
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply core/adapters/docker/client_internal_test.go:121]: Addressed in commit bd8e77a: switched nil-config/no-writer tests to errors.Is on the new sentinels, extracted newLoopbackSDKClient helper to drop the duplicate SDK-setup boilerplate flagged by SonarCloud, and added a CHANGELOG entry for the two new sentinel errors.
